@nx/jest resolves ts paths wrong

### Current Behavior

In my project's tsconfig.json and tsconfig.spec.json I specified alias paths

```
  "paths": {
      "~~*": ["../../*"], //<-- refers to the workspace's root
     // .... other paths
    }
```
in <my-app>/jest.config.ts

```
import jestConfig from '~~jest.config';
export default jestConfig({ _projectRoot: __dirname });
```

When I add the plugin `nx add @nx/jest`, the following error occurs
```
5 errors occurred while processing files for the @nx/jest/plugin plugin
.
  - apps/assets/jest.config.ts: Cannot find module '../../~~jest.config'
Require stack:
- /workspaces/mashrouk/apps/assets/jest.config.ts
- /workspaces/mashrouk/node_modules/.pnpm/@nx+devkit@21.1.2_nx@21.1.2_@swc-node+register@1.10.10_@swc+core@1.11.29_@swc+helpers@0.5.17__ffljwscjdaomkqev3az6ugtrei/node_modules/@nx/devkit/src/utils/config-utils.js
- /workspaces/mashrouk/node_modules/.pnpm/@nx+jest@21.1.2_@babel+traverse@7.27.1_@swc-node+register@1.10.10_@swc+core@1.11.29_@swc+help_oj4j7lrynaepz4txa3cmv4ytzy/node_modules/@nx/jest/src/plugins/plugin.js
- /workspaces/mashrouk/node_modules/.pnpm/@nx+jest@21.1.2_@babel+traverse@7.27.1_@swc-node+register@1.10.10_@swc+core@1.11.29_@swc+help_oj4j7lrynaepz4txa3cmv4ytzy/node_modules/@nx/jest/src/generators/init/init.js
- /workspaces/mashrouk/node_modules/.pnpm/nx@21.1.2_@swc-node+register@1.10.10_@swc+core@1.11.29_@swc+helpers@0.5.17__@swc+types@0.1.21_7t2fh5y3gx2kwbf7qy6cwswh3m/node_modules/nx/src/config/schema-utils.js
- /workspaces/mashrouk/node_modules/.pnpm/nx@21.1.2_@swc-node+register@1.10.10_@swc+core@1.11.29_@swc+helpers@0.5.17__@swc+types@0.1.21_7t2fh5y3gx2kwbf7qy6cwswh3m/node_modules/nx/src/command-line/run/executor-utils.js
- /workspaces/mashrouk/node_modules/.pnpm/nx@21.1.2_@swc-node+register@1.10.10_@swc+core@1.11.29_@swc+helpers@0.5.17__@swc+types@0.1.21_7t2fh5y3gx2kwbf7qy6cwswh3m/node_modules/nx/src/project-graph/utils/project-configuration-utils.js
- /workspaces/mashrouk/node_modules/.pnpm/nx@21.1.2_@swc-node+register@1.10.10_@swc+core@1.11.29_@swc+helpers@0.5.17__@swc+types@0.1.21_7t2fh5y3gx2kwbf7qy6cwswh3m/node_modules/nx/src/utils/package-json.js
- /workspaces/mashrouk/node_modules/.pnpm/nx@21.1.2_@swc-node+register@1.10.10_@swc+core@1.11.29_@swc+helpers@0.5.17__@swc+types@0.1.21_7t2fh5y3gx2kwbf7qy6cwswh3m/node_modules/nx/bin/nx.js
  - apps/core/jest.config.ts: Cannot find module '../../~~jest.config'
...
```
According to ts paths, the resolved path should be `../../jest.config` instead of `../../~~jest.config`

I can run the test without NX without any problem, and vs code can see the path correctly

![Image](https://github.com/user-attachments/assets/6e94f931-f0bf-4caf-bfcd-fa8b0816335e)

### Expected Behavior

The alias path should be resolved correctly

### GitHub Repo

_No response_

### Steps to Reproduce

1. add paths to the projects tsconfig.json and tsconfig.spec.json
2. create jest.config.ts in the project's root
3. use the alias path


### Nx Report

```shell
Node           : 22.12.0
OS             : linux-x64
Native Target  : x86_64-linux
pnpm           : 9.15.0

nx (global)  : 21.1.2
nx           : 21.1.2
@nx/js       : 21.1.2
@nx/jest     : 21.1.2
@nx/eslint   : 21.1.2
typescript   : 5.8.3
---------------------------------------
Registered Plugins:
@nx/js/typescript
---------------------------------------
Cache Usage: 7.76 MB / 46.73 GB
```

### Failure Logs

```shell

```

### Package Manager Version

pnpm v9.15.0

### Operating System

- [ ] macOS
- [x] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/packages/devkit/src/utils/config-utils.ts
+++ b/packages/devkit/src/utils/config-utils.ts
@@ -10,16 +10,21 @@ export let dynamicImport = new Function(
 );
 
 export async function loadConfigFile<T extends object = any>(
-  configFilePath: string
+  configFilePath: string,
+  tsconfigFileNames?: string[]
 ): Promise<T> {
   const extension = extname(configFilePath);
-  const module = await loadModule(configFilePath, extension);
+  const module = await loadModule(configFilePath, extension, tsconfigFileNames);
   return module.default ?? module;
 }
 
-async function loadModule(path: string, extension: string): Promise<any> {
+async function loadModule(
+  path: string,
+  extension: string,
+  tsconfigFileNames?: string[]
+): Promise<any> {
   if (isTypeScriptFile(extension)) {
-    return await loadTypeScriptModule(path, extension);
+    return await loadTypeScriptModule(path, extension, tsconfigFileNames);
   }
   return await loadJavaScriptModule(path, extension);
 }
@@ -30,9 +35,10 @@ function isTypeScriptFile(extension: string): boolean {
 
 async function loadTypeScriptModule(
   path: string,
-  extension: string
+  extension: string,
+  tsconfigFileNames?: string[]
 ): Promise<any> {
-  const tsConfigPath = getTypeScriptConfigPath(path);
+  const tsConfigPath = getTypeScriptConfigPath(path, tsconfigFileNames);
 
   if (tsConfigPath) {
     const unregisterTsProject = registerTsProject(tsConfigPath);
@@ -46,10 +52,16 @@ async function loadTypeScriptModule(
   return await loadModuleByExtension(path, extension);
 }
 
-function getTypeScriptConfigPath(path: string): string | null {
+function getTypeScriptConfigPath(
+  path: string,
+  tsconfigFileNames?: string[]
+): string | null {
   const siblingFiles = readdirSync(dirname(path));
-  return siblingFiles.includes('tsconfig.json')
-    ? join(dirname(path), 'tsconfig.json')
+  const tsConfigFileName = (tsconfigFileNames ?? ['tsconfig.json']).find(
+    (name) => siblingFiles.includes(name)
+  );
+  return tsConfigFileName
+    ? join(dirname(path), tsConfigFileName)
     : getRootTsConfigPath();
 }
 

--- a/packages/jest/src/plugins/plugin.ts
+++ b/packages/jest/src/plugins/plugin.ts
@@ -249,7 +249,16 @@ async function buildJestTargets(
   const absConfigFilePath = resolve(context.workspaceRoot, configFilePath);
 
   if (require.cache[absConfigFilePath]) clearRequireCache();
-  const rawConfig = await loadConfigFile(absConfigFilePath);
+  const rawConfig = await loadConfigFile(
+    absConfigFilePath,
+    // lookup for the same files we look for in the resolver and fall back to tsconfig.json
+    [
+      'tsconfig.spec.json',
+      'tsconfig.test.json',
+      'tsconfig.jest.json',
+      'tsconfig.json',
+    ]
+  );
 
   const targets: Record<string, TargetConfiguration> = {};
   const namedInputs = getNamedInputs(projectRoot, context);

